flow360/component/project.py

```diff
@@ -0,0 +1,579 @@
+        Flow360ValueError
+            If the root asset cannot be retrieved for the project.
+        """
+        project_api = RestApi(ProjectInterface.endpoint, id=project_id)
+        info = project_api.get()
+        if not isinstance(info, dict):
+            raise Flow360WebError(f"Cannot load project {project_id}, missing project data.")
+        if not info:
+            raise Flow360WebError(f"Couldn't retrieve project info for {project_id}")
+        meta = ProjectMeta(**info)
+        root_asset = None
+        root_type = meta.root_item_type
+        if root_type == RootType.GEOMETRY:
+            root_asset = Geometry.from_cloud(meta.root_item_id)
+        elif root_type == RootType.VOLUME_MESH:
+            root_asset = VolumeMeshV2.from_cloud(meta.root_item_id)
+        project = Project(metadata=meta, solver_version=root_asset.solver_version)
+        project._project_webapi = project_api
+        if not root_asset:
+            raise Flow360ValueError(f"Couldn't retrieve root asset for {project_id}")
+        if root_type == RootType.GEOMETRY:
+            project._geometry = root_asset
+            project._root_webapi = RestApi(GeometryInterface.endpoint, id=root_asset.id)
+        elif root_type == RootType.VOLUME_MESH:
+            project._volume_mesh = root_asset
+            project._root_webapi = RestApi(VolumeMeshInterfaceV2.endpoint, id=root_asset.id)
+        project._get_root_simulation_json()
+        return project
+
+    def _check_initialized(self):
+        """
+        Checks if the project instance has been initialized correctly.
+
+        Raises
+        ------
+        Flow360ValueError
+            If the project is not initialized.
+        """
+        if not self.metadata or not self.solver_version:
+            raise Flow360ValueError(
+                "Project is not initialized - use Project.from_file or Project.from_cloud"
+            )
+
+    def _get_root_simulation_json(self):
+        """
+        Loads the default simulation JSON for the project based on the root asset type.
+
+        Raises
+        ------
+        Flow360ValueError
+            If the root item type or ID is missing from project metadata.
+        Flow360WebError
+            If the simulation JSON cannot be retrieved.
+        """
+        self._check_initialized()
+        root_type = self.metadata.root_item_type
+        root_id = self.metadata.root_item_id
+        if not root_type or not root_id:
+            raise Flow360ValueError("Root item type or ID is missing from project metadata")
+        resp = self._root_webapi.get(method="simulation/file", params={"type": "simulation"})
+        if not isinstance(resp, dict) or "simulationJson" not in resp:
+            raise Flow360WebError("Couldn't retrieve default simulation JSON for the project")
+        simulation_json = json.loads(resp["simulationJson"])
+        self._root_simulation_json = simulation_json
```

Comment:
Ideally we can have` _root_simulation_json` as a property function and get the JSON in that function if cached JSON does not exist. But we do not have to for now since the code path is pretty simple.